alfasim_sdk_api/detail/bootstrap_linux.h

```diff
@@ -10,74 +10,69 @@
-        get_plugin_variable = (get_plugin_variable_func)dlsym(this->m_handle, "get_plugin_variable");
-        get_field_id = (get_field_id_func)dlsym(this->m_handle, "get_field_id");
-        get_primary_field_id_of_phase = (get_field_id_func)dlsym(this->m_handle, "get_primary_field_id_of_phase");
-        get_phase_id = (get_phase_id_func)dlsym(this->m_handle, "get_phase_id");
-        get_layer_id = (get_layer_id_func)dlsym(this->m_handle, "get_layer_id");
-        get_state_variable_array = (get_state_variable_array_func)dlsym(this->m_handle, "get_state_variable_array");
-        get_simulation_array = (get_simulation_array_func)dlsym(this->m_handle, "get_simulation_array");
-        get_simulation_tracer_array = (get_simulation_tracer_array_func)dlsym(this->m_handle, "get_simulation_tracer_array");
-        get_simulation_quantity = (get_simulation_quantity_func)dlsym(this->m_handle, "get_simulation_quantity");
-        get_wall_interfaces_temperature = (get_wall_interfaces_temperature_func)dlsym(this->m_handle, "get_wall_interfaces_temperature");
-        get_flow_pattern = (get_flow_pattern_func)dlsym(this->m_handle, "get_flow_pattern");
-        get_plugin_input_data_table_quantity = (get_plugin_input_data_table_quantity_func)dlsym(this->m_handle, "get_plugin_input_data_table_quantity");
-        get_tracer_id = (get_tracer_id_func)dlsym(this->m_handle, "get_tracer_id");
-        get_tracer_name_size = (get_tracer_name_size_func)dlsym(this->m_handle, "get_tracer_name_size");
-        get_tracer_name = (get_tracer_name_func)dlsym(this->m_handle, "get_tracer_name");
-        get_tracer_ref_by_name = (get_tracer_ref_by_name_func)dlsym(this->m_handle, "get_tracer_ref_by_name");
-        get_tracer_partition_coefficient = (get_tracer_partition_coefficient_func)dlsym(this->m_handle, "get_tracer_partition_coefficient");
-        get_wall_layer_id = (get_wall_layer_id_func)dlsym(this->m_handle, "get_wall_layer_id");
-        set_wall_layer_property = (set_wall_layer_property_func)dlsym(this->m_handle, "set_wall_layer_property");
-        get_plugin_input_data_multiplereference_selected_size = (get_plugin_input_data_multiplereference_selected_size_func)dlsym(this->m_handle, "get_plugin_input_data_multiplereference_selected_size");
+    // Find the alfasim installation path: Prefer the environment variable ALFASIM_PATH, but, if not found,
+    // try the same path as the executable
+    char *alfasim_executable_dir;
+    if (executable_dir_from_env != NULL) {
+        alfasim_executable_dir = executable_dir_from_env;
+    } else {
+        alfasim_executable_dir = current_exe_dir;
     }
 
-    ~alfasim_sdk_bootstrap()
-    {
-        dlclose(this->m_handle);
-    }
+    // Load shared object
+    char *full_filepath = NULL;
+    asprintf(&full_filepath, "%s%s", alfasim_executable_dir, SO_FILENAME);
+    api->handle = dlopen(full_filepath, RTLD_LAZY);
```

Comment:
Need to `free(full_filepath);`

Reply:
And also `alfasim_executable_dir`.

Reply:
Actually, alfasim_executable_dir resides on the stack or on the system's memory (when it comes from `getenv`), so it doesn't need any free

Reply:
OK!